exercises/complex-numbers/examples/success-standard/src/ComplexNumbers.hs

```diff
@@ -5,34 +5,46 @@ module ComplexNumbers
  mul,
  add,
  sub,
- div) where
+ div,
+ real,
+ imaginary,
+ complex) where
 
 import Prelude hiding (div, abs)
 
 -- Data definition -------------------------------------------------------------
-data Complex = Complex {real :: Float, imaginary :: Float} deriving (Eq, Show)
+data Complex a = Complex a a deriving (Eq, Show)
 
 --------------------------------------------------------------------------------
-exp2 :: Float -> Float
-exp2 a = a ^ (2 :: Integer)
+exp2 :: Num a => a -> a
+exp2 x = x^(2 :: Integer)
 
 -- unary operators -------------------------------------------------------------
-conjugate :: Complex -> Complex
+real :: Num a => Complex a -> a
+real (Complex a _) = a
+
+imaginary :: Num a => Complex a -> a
+imaginary (Complex _ b) = b
+
+conjugate :: Num a => Complex a -> Complex a
 conjugate (Complex a b) = Complex a (-b)
 
-abs :: Complex -> Float
+abs :: Floating a => Complex a -> a
 abs (Complex a b) = sqrt (exp2 a + exp2 b)
 
 -- binary operators ------------------------------------------------------------
-mul :: Complex -> Complex -> Complex
+complex :: (a, a) -> Complex a
+complex (a, b) = Complex a b
```

Comment:
What about `complex = curry Complex` ?

Reply:
Oh sorry, that needs to be `complex = uncurry Complex`, get them mixed up too often.

Reply:
Forgot about that. I don't use neither of them too often.